code/modules/projectiles/projectile/ego_bullets/waw.dm

```diff
@@ -44,9 +44,33 @@
 
 /obj/projectile/ego_bullet/ego_magicbullet
 	name = "magic bullet"
+	icon_state = "magic_bullet"
 	damage = 80
 	speed = 0.1
 	damage_type = BLACK_DAMAGE
 	flag = BLACK_DAMAGE
 	projectile_piercing = PASSMOB
 	projectile_phasing = (ALL & (~PASSMOB) & (~PASSCLOSEDTURF))
+
+//Smartgun
+/obj/projectile/ego_bullet/ego_loyalty
+	name = "loyalty"
+	icon_state = "loyalty"
+	damage = 8
+	speed = 0.2
+	nodamage = TRUE	//Damage is calculated later
+	damage_type = RED_DAMAGE
+	flag = RED_DAMAGE
+	projectile_piercing = PASSMOB
+	projectile_phasing = (ALL & (~PASSMOB) & (~PASSCLOSEDTURF))
+
+/obj/projectile/ego_bullet/ego_loyalty/on_hit(atom/target, blocked = FALSE)
+	if(!ishuman(target))
+		nodamage = FALSE
+	else
```

Comment:
I hope you tested it, because as far as I can tell, this is not supposed to work.

Reply:
Egor, Listen. I was testing it too.
IT DOES WORK.
I showed it off working to like 6 different people

Reply:
It passes through people, and if it hits a non-human it deals damage and deletes itself.